src/edu/csus/ecs/pc2/ui/AutoJudgingMonitor.java

```diff
@@ -338,8 +346,15 @@ public void runChanged(RunEvent event) {
                         // and we received a not available for the run we were requesting
                         if (event.getRun().getNumber() ==  runBeingAutoJudged.getNumber() 
                                 && event.getRun().getSiteNumber() == runBeingAutoJudged.getSiteNumber()) {
-                            // XXX should we log this?
-                            // claim to have received it, so fetchRun can exit
+                            
+                            // the run we requested is not available -- log this unusual (though possibly legitimate) condition
+                            log.info("Received 'NOT_AVAILABLE' message for requested run " + event.getRun().getNumber() );
+                            
+                            //add the run to the list of requested-but-unavailable runs 
+                            // (a patch to support keeping the AJ from continually re-requesting the same run; see https://github.com/pc2ccs/pc2v9/issues/480)
+                            getUnavailableRunsList().addRun(runBeingAutoJudged); 
```

Comment:
Yes.  Agreed, but some of the exceptions thrown by addRun currently do a System.err.println().  If a try/catch/log is added in AutoJudingMonitor, these System.err.println's should be removed since the content of the exception message is identical to the System.err.println.

Reply:
re: exception handling
I agree.   We should work to avoid System.err and instead either write to a log or throw an exception.
Note that the System.err in modern loggers has been replaced by adding a console appender to the log instance.
And pc2 has a console appender/handler.   In the Log method is a method to start a console handler
log.startConsoleLogger():   That will write out log messages to console and there is also a complimentary
log.stopConsoleLogger()
Lastly if debug output is for stderr or stdout, that is where my "debug 22" came from, find another phrase
that indicates that the stderr/stdout output should be removed before the final git commit.
And likewise if stopConsoleLogger or startConsoleLogger are used for debugging those should removed before
the final commit.